src/Application.vala

```diff
@@ -0,0 +1,202 @@
+using Gtk;
+
+namespace Ilia {
+    class Application : Gtk.Application {
+        private GLib.HashTable<string, string?> arg_map;
+
+        public Application () {
+            Object (
+              application_id: "org.regolith.launcher",
+              flags: ApplicationFlags.REPLACE | ApplicationFlags.ALLOW_REPLACEMENT
+            );
+        }
+
+        protected override bool local_command_line (ref unowned string[] args, out int exit_status) {
+            this.arg_map = parse_args (args);
+            if (arg_map.contains ("-h") || arg_map.contains ("--help")) print_help_and_exit ();
+            if (arg_map.contains ("-v") || arg_map.contains ("--version")) print_version_and_exit ();
+            args[0] = null;
+            return base.local_command_line (ref args, out exit_status);
+        }
+
+        protected override void activate () {
+            var window = new Ilia.DialogWindow (this.arg_map);
+            window.set_application (this);
+
+            // Grab inputs from wayland backend before showing window
+            if (IS_SESSION_WAYLAND) {
+                GtkLayerShell.init_for_window (window);
```

Comment:
note: it's interesting that these methods are apparently infallible.  Is an exception thrown or something in the case of some error (memcpy, etc)

Reply:
I'm actually not sure if the are. However, there's no clear indication in the documentation for this library about any exceptions. Maybe they are already handled?

Reply:
Maybe!  Yeah the gtk/gdk docs coverage has always been spotty.